django/core/serializers/python.py

```diff
@@ -63,6 +63,8 @@ def handle_fk_field(self, obj, field):
                 value = None
         else:
             value = getattr(obj, field.get_attname())
```

Comment:
I think we can use `if not is_protected_type(value):` here. Then the changes in the m2m tests can be reverted and this should be good to go.